project/Win32BuildSetup/genNsisInstaller.nsi

```diff
@@ -254,23 +283,85 @@ Section -StartMenu
-     "" "$INSTDIR\Uninstall.exe" 0 SW_SHOWNORMAL \
-     "" "Uninstall ${APP_NAME}."
-  CreateShortCut "$SMPROGRAMS\${APP_NAME}.lnk" "$INSTDIR\${APP_NAME}.exe" \
-     "" "$INSTDIR\${APP_NAME}.exe" 0 SW_SHOWNORMAL \
-     "" "Start ${APP_NAME}."
+
+  ${If} $SILENT_AND_FORCEINSTALL = 1 
+    ${If} $FORCEINSTALL_ADD_START_MENU_LINK = 1
+      LogEx::Write "Adding start menu link under silent force install"
+      CreateDirectory "$SMPROGRAMS\$StartMenuFolder"
+      CreateShortCut "$SMPROGRAMS\$StartMenuFolder\${APP_NAME}.lnk" "$INSTDIR\${APP_NAME}.exe" \
+         "" "$INSTDIR\${APP_NAME}.exe" 0 SW_SHOWNORMAL \
+         "" "Start ${APP_NAME}."
+
+      CreateShortCut "$SMPROGRAMS\${APP_NAME}.lnk" "$INSTDIR\${APP_NAME}.exe" \
+         "" "$INSTDIR\${APP_NAME}.exe" 0 SW_SHOWNORMAL \
+         "" "Start ${APP_NAME}."
+    ${EndIf}
+  ${Else}
+    CreateDirectory "$SMPROGRAMS\$StartMenuFolder"
+    CreateShortCut "$SMPROGRAMS\$StartMenuFolder\${APP_NAME}.lnk" "$INSTDIR\${APP_NAME}.exe" \
+       "" "$INSTDIR\${APP_NAME}.exe" 0 SW_SHOWNORMAL \
+       "" "Start ${APP_NAME}."
+   CreateShortCut "$SMPROGRAMS\${APP_NAME}.lnk" "$INSTDIR\${APP_NAME}.exe" \
+       "" "$INSTDIR\${APP_NAME}.exe" 0 SW_SHOWNORMAL \
+       "" "Start ${APP_NAME}."
+  ${EndIf}
+
+
+  ${If} $SILENT_AND_FORCEINSTALL = 1 
+    ${If} $FORCEINSTALL_CREATE_UNINSTALLER_SETTINGS = 1
+      ${AndIf} $FORCEINSTALL_ADD_START_MENU_LINK = 1
+        LogEx::Write "Adding uninstaller link in start menu under silent force install"
+        CreateShortCut "$SMPROGRAMS\$StartMenuFolder\Uninstall ${APP_NAME}.lnk" "$INSTDIR\Uninstall.exe" \
+           "" "$INSTDIR\Uninstall.exe" 0 SW_SHOWNORMAL \
+           "" "Uninstall ${APP_NAME}."
+    ${EndIf}
+  ${Else}
+    CreateShortCut "$SMPROGRAMS\$StartMenuFolder\Uninstall ${APP_NAME}.lnk" "$INSTDIR\Uninstall.exe" \
+       "" "$INSTDIR\Uninstall.exe" 0 SW_SHOWNORMAL \
+       "" "Uninstall ${APP_NAME}."
+  ${EndIf}
+
+  ${If} $SILENT_AND_FORCEINSTALL = 1 
+    ${If} $FORCEINSTALL_ADD_QUICKLAUNCH_LINK = 1
+      LogEx::Write "Adding quick launch link under silent force install"
+      CreateShortCut "$QUICKLAUNCH\${APP_NAME}.lnk" "$INSTDIR\${APP_NAME}.exe" \
+         "" "$INSTDIR\${APP_NAME}.exe" 0 SW_SHOWNORMAL \
+         "" "Start ${APP_NAME}."
+    ${EndIf}
+  ${Else}
+    CreateShortCut "$QUICKLAUNCH\${APP_NAME}.lnk" "$INSTDIR\${APP_NAME}.exe" \
+       "" "$INSTDIR\${APP_NAME}.exe" 0 SW_SHOWNORMAL \
+       "" "Start ${APP_NAME}."
+  ${EndIf}
 
   ; Desktop Shortcut
-  File "${ICON}"
-  createShortCut "$DESKTOP\${APP_NAME}.lnk" "$INSTDIR\${START_EXE}" "" "$INSTDIR\application.ico"
+  ${If} $SILENT_AND_FORCEINSTALL = 1 
+    ${If} $FORCEINSTALL_ADD_DESKTOP_LINK = 1
+      LogEx::Write "Adding desktop shortcut under silent force install"
+      File "${ICON}"
+      createShortCut "$DESKTOP\${APP_NAME}.lnk" "$INSTDIR\${START_EXE}" "" "$INSTDIR\application.ico"
```

Comment:
minor thing but we should use either ${APP_NAME}.exe or ${START_EXE} but not both.

Reply:
`APP_NAME = Play`
`START_EXE = Play.exe`

In this case we want to keep things the way they are since we don't want a Desktop shortcut to look like `Play.exe.lnk`.

Reply:
I'm suggesting getting rid of ${START_EXE} and just using ${APP_NAME}.exe like it is used in other lines. but again this is minor comment

Reply:
Ahhh okay gotcha.